decidim-core/lib/decidim/commands/update_resource.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Commands
+    class UpdateResource < ::Decidim::Command
+      include Decidim::Commands::ResourceHandler
+
+      # Initializes the command.
+      #
+      # @param form [Decidim::Form] the form object to update the resource.
+      # @param resource [Decidim::Resource] the resource to update.
+      def initialize(form, resource)
+        @form = form
+        @resource = resource
+      end
+
+      # Updates the timeline_entry if valid.
+      #
+      # Broadcasts :ok if successful, :invalid otherwise.
+      def call
+        return broadcast(:invalid) if invalid?
+
+        transaction do
+          run_before_hooks
+          update_resource
+          run_after_hooks
+        end
+
+        broadcast(:ok, resource)
+      rescue Decidim::Commands::HookError
+        broadcast(:invalid)
+      end
+
+      protected
+
+      attr_reader :form, :resource
+
+      def update_resource
+        Decidim.traceability.update!(
+          resource,
+          form.current_user,
+          attributes,
+          **extra_params
+        )
+      end
+
+      # Runs any before hooks. That you may want to execute before updating the resource.
```

Comment:
This wording is weird. How about something like this?
```suggestion
      # Useful for running any code that you may want to execute before updating the resource.
```

We should also change it in the other comments

Reply:
done in 94771d78819c3002354367f20f75b28d814444b9